### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by arturo-seijas**
Load proxy configuration from the juju model env vars

### CODE REVIEW

File: src/charm_state.py [hunk 1]
```diff
@@ -16,6 +18,37 @@
 )
 
 
+class ProxyConfig(BaseModel):
```
Comment (jdkandersson) [inline src/charm_state.py:21]: What is the reason for not making this part of the `CharmState` class?

Comment (arturo-seijas) [reply src/charm_state.py:21]: None in particular. Are you suggesting that to improve consistency with other charms?

Comment (jdkandersson) [reply src/charm_state.py:21]: Well that and why not just make it part of the charm state?

Comment (amandahla) [reply src/charm_state.py:21]: @jdkandersson @gregory-schiano Did you came to a decision about that as mentioned here?
https://github.com/canonical/flask-k8s-operator/pull/33#discussion_r1271782149

Comment (jdkandersson) [reply src/charm_state.py:21]: Not yet, I'll discuss with gregory today

Comment (jdkandersson) [reply src/charm_state.py:21]: Actually we did discuss it, it should go into the state

Comment (arturo-seijas) [reply src/charm_state.py:21]: Done

File: tests/unit/test_charm_state.py [hunk 2]
```diff
@@ -23,36 +49,30 @@ def test_known_charm_config():
...
     """
-    harness = Harness(SamlIntegratorOperatorCharm)
-    harness.begin()
+    charm = MagicMock(config={})
     try:
-        CharmState.from_charm(harness.charm)
+        CharmState.from_charm(charm)
         assert False
     except CharmConfigInvalidError:
         assert True
```
Comment (yanksyoon) [inline tests/unit/test_charm_state.py:78]: Would this be better with `with pytest.raises`?

**Review by yanksyoon — APPROVED** [review]
LGTM, minor suggestions!

File: src/charm.py [hunk 3]
```diff
@@ -28,27 +29,29 @@ def __init__(self, *args):
...
         """Install needed apt packages."""
+        # The apt lib uses the env vars to set the proxy configuration
         self.unit.status = ops.MaintenanceStatus("Installing packages")
-        apt.update()
-        apt.add_package(["libxml2"])
+        proxy_config = self._charm_state.proxy_config
+        os.environ["HTTP_PROXY"] = proxy_config.http_proxy
+        os.environ["HTTPS_PROXY"] = proxy_config.https_proxy
+        os.environ["NO_PROXY"] = proxy_config.no_proxy
+        apt.add_package(["libxml2"], update_cache=True)
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply src/charm.py:50]: Good to know. I'll close the PR then

[the remark this answers is not in the document]
Comment (weiiwang01) [reply src/charm.py:50]: Maybe we still need the http proxy to access the metadata url?

**Comment by jdkandersson** [comment]
Is this still needed?

**Comment by arturo-seijas** [comment]
> Is this still needed?

No, juju will atuomatically inject configuration files for apt and this is the only use case for the proxy configuration in this charm

**Pull request closed without merging**